### ISSUE TITLE
How We Work

### ISSUE THREAD

**Issue by jel535s** [MidwestAccessCoalition/lockbox_rails#408]
## This GitHub
The project board is laid out in a somewhat kanban style, with the columns to the left being tickets that are being spec’d out by the project management team.  The two main columns you’ll be working in are more to the right: “To Do” and “In Progress”.  More detail is below about how to pick up and finish tickets in these columns.

## Our Weekly Meetings
Each Sunday at 4pm, we start with a standup where each volunteer discusses what they are currently working on, any reviews that are needed, and anything that is a block currently.  

The rest of the meeting tends to be fairly heads-down work, although we often share jokes or ask questions regarding the issues we are working on.  It is normal to wear headphones during the later part of the meeting, but it’s also okay to Slack someone a question if you need a second opinion, even if they are dialed-in to their work at the moment.  We are all here to help each other out!  We usually end the meeting just checking in to make sure no one is blocked by anything.

## Outside The Sunday Meeting
As bandwidth allows, some folks do work on their tickets outside of the meeting.  This is absolutely wonderful, if you have the energy and time to do so!  It might be that some weeks you have an extra hour or even an evening to push some code- other weeks you may not.  

We are active on Slack and we do meet up socially sometimes as well.  Most of the group has been working together now for a while and we’re always happy to have new members like you!  A good place to start with our Slack is the #tech channel.  You will want to check that channel every Sunday morning, to make sure you know the location where we are meeting.  You’ll find lots of other channels are active as well though.  The expectation on Slack DMs is that you’d respond as soon as quickly as you can, without interrupting anything important.  

## Changing Tickets
If you don’t have any tickets, during our weekly stand up, we do self-assignment to any open tickets on the “To Do” column.  Since this is a fully volunteer group, we want you to work as much as possible on items that are interesting to you, or that fit your skills well, or that are a useful challenge.  The project management team can also help advise you on what to pick up, or on what is the most urgent need, so don’t hesitate to ask!  

When you pick up a ticket, there are 3 steps.

1. Open the ticket in GitHub and assign yourself. Let’s practice that right now!  On the right, notice the area that says “Assignees”.  Click the gear icon, and find your name.  Click on your name to assign this ticket to yourself.
2. Then, change the label.  On a regular ticket, you’ll change it from “Ready to work” to “WIP”.  On this ticket, again look at the right side and under “Labels” select the gear icon and choose the “needs spec” label.
3. Finally, back on the main GitHub board, drag the ticket from the “To Do” column into the “In Progress” column.  For this ticket, drag it from the “Documentation” tab all the way on the right to the “Requires spec/review” column closer to the left of the board.

If you pick up a ticket but it turns out it isn’t a good fit, please say so earlier rather than later.  It’s totally fine to give back a ticket, even mid-meeting.  Ideally you’d document what you’ve already tried on the ticket itself, before notifying the project management team.  Then remove yourself as the assignee.  Change the label from “WIP” to “ready to work”- or if the problem is that the ticket isn’t fully spec’d out, you might change the label to “needs spec”.  Finally, move the ticket from the “In Progress” column back to the “To Do” column, so that another developer can pick it up.

## Completing a Ticket
You finished a branch and a PR is ready for review- Hooray!  There are a couple things to do just to close the loop for the rest of the team.

When your PR is ready for review, please label it “ready for review”.  You can assign that review to certain folks if you know it’s in their wheelhouse or related to what they’ve recently worked on, but you don’t need to assign a reviewer.  Then, please go to the ticket in GitHub and also change that label to “ready for review”, and move it to the “In Review” column.  That gives our reviewers a heads up to get to it asap.

Once your ticket is in review, you are free to self-assign a new ticket from the “To Do” board.  

The reviewer may ask you to change a few things or run some other tests, and they’ll tag you in the PR for that.  Keep an eye out for Slack DMs, or emails from the team with a subject that contains “Re:[MidwestAccessCoalition/lockbox_rails]”.  

When your review has been passed, the ticket will be closed by the reviewer.  

## Documenting New Issues
If you come across a new issue that we don’t have a ticket for yet, or even if you’re just not sure, alert someone on the project management team- ideally, share screenshots!  They’ll either write up a ticket or incorporate it into an existing ticket, depending on the issue. As you get more comfortable with how we work, you may start writing up your own tickets.  Please have the project management team look these over to make sure they match the tone and process we usually follow.  

To create a ticket, the easiest way is to use the green “New Issue” button at the top right hand of this page.  Go ahead and right-click on that now, to open it in a new tab.  Instead of actually creating a new issue at this time, we will close this out together.  You’ll see a new tab with a template ticket shows up there.  On the template ticket, on the right side on the page, you’ll see a “Projects” section.  Select the “Lockbox MVP Release” project.  Create some test text, then click the green “Submit New Issue” button.  On the new issue page, scroll down and enter in a comment that this was just a test issue, then click the red “close issue” button.
